### PR TITLE
Content Security Policy Cheat Sheet with more info

### PR DESCRIPTION
**Pull request by shaialon**
Thank you for submitting a Pull Request (PR) to the Cheat Sheet Series. 

> :triangular_flag_on_post: If your PR is related to grammar/typo mistakes, please double-check the file for other mistakes in order to fix all the issues in the current cheat sheet.

Please make sure that for your contribution:

- [ ] In case of a new Cheat Sheet, you have used the [Cheat Sheet template](https://github.com/OWASP/CheatSheetSeries/blob/master/templates/New_CheatSheet.md).
- [ ] All the markdown files do not raise any validation policy violation, see the [policy](https://github.com/OWASP/CheatSheetSeries/actions?query=workflow%3A%22Markdown+Link+Check%22).
- [ ] All the markdown files follow these [format rules](https://github.com/OWASP/CheatSheetSeries/blob/master/CONTRIBUTING.md#markdown).
- [ ] All your assets are stored in the **assets** folder.
- [ ] All the images used are in the **PNG** format.
- [ ] Any references to websites have been formatted as [TEXT](URL)
- [ ] You verified/tested the effectiveness of your contribution (e.g., the defensive code proposed is really an effective remediation? Please verify it works!).
- [ ] The CI build of your PR pass, see the build status [here](https://github.com/OWASP/CheatSheetSeries/actions).

If your PR is related to an issue, please finish your PR text with the following line:

This PR covers issue #<insert number here>.

Thank you again for your contribution :smiley:

### CODE REVIEW

File: cheatsheets/Content_Security_Policy_Cheat_Sheet.md
Comment (jmanico) [inline cheatsheets/Content_Security_Policy_Cheat_Sheet.md:84]: Can you link this to https://caniuse.com/?search=navigate-to as a reference?

File: cheatsheets/Content_Security_Policy_Cheat_Sheet.md
Comment (jmanico) [inline cheatsheets/Content_Security_Policy_Cheat_Sheet.md:196]: unsigned? I do not believe digital signatures are a part of CSP

**Review by jmanico — CHANGES_REQUESTED** [review]
Can you just link up caniuse for navigate-to and clarify the use of the term unsigned?

File: cheatsheets/Content_Security_Policy_Cheat_Sheet.md
Comment (jmanico) [reply cheatsheets/Content_Security_Policy_Cheat_Sheet.md:196]: Suggest changing this to:

When `default-src` or `script-src*` directives are active, CSP by default disables any JavaScript code placed inline in the HTML source, such as this:

And can you also please remove the other use of the term signed in this document?

Comment (shaialon) [reply cheatsheets/Content_Security_Policy_Cheat_Sheet.md:196]: I believe this references two main ways of "signing" code with CSP:
1. `nonce` 
2. SHA hashes (as describes right after.

We could change to 
```suggestion
When `default-src` or `script-src*` directives are active, CSP by default disables any inline JavaScript code, specified as script-elements or script-attributes, such as this:
```

Comment (jmanico) [reply cheatsheets/Content_Security_Policy_Cheat_Sheet.md:196]: signed in the wrong term. Here is the exact text from the CSP 2 standard to consider.

The script-src directive lets developers specify exactly which script elements on a page were intentionally included for execution. Ideally, developers would avoid inline script entirely and whitelist scripts by URL. However, in some cases, removing inline scripts can be difficult or impossible. For those cases, developers can whitelist scripts using a randomly generated nonce.

and

The script-src directive lets developers whitelist a particular inline script by specifying its hash as an allowed source of script.

Comment (jmanico) [reply cheatsheets/Content_Security_Policy_Cheat_Sheet.md:196]: And here is similar text from the CSP 3 standard to consider

The script-src directive governs four things:

Script requests MUST pass through §4.1.3 Should request be blocked by Content Security Policy?.

Script responses MUST pass through §4.1.4 Should response to request be blocked by Content Security Policy?.

Inline script blocks MUST pass through §4.2.3 Should element’s inline type behavior be blocked by Content Security Policy?. Their behavior will be blocked unless every policy allows inline script, either implicitly by not specifying a script-src (or default-src) directive, or explicitly, by whitelisting "unsafe-inline", a nonce-source or a hash-source that matches the inline block.

The following JavaScript execution sinks are gated on the "unsafe-eval" source expression:

eval()

Function()

setTimeout() with an initial argument which is not callable.

setInterval() with an initial argument which is not callable.

Comment (jmanico) [reply cheatsheets/Content_Security_Policy_Cheat_Sheet.md:196]: Sincerely, there are zero mentions of digial signatures or similar in the CSP standard. =)

**Review by mackowski — APPROVED** [review]
Looks good! Thank you!

**Comment by shaialon** [comment]
@jmanico , are we good to merge?
(it seems to be stuck pending your approval...)
Thanks.
image

**Review by jmanico — COMMENTED** [review]
Great work!

**Pull request merged**
